opnode/test/system_test.go

```diff
@@ -125,10 +128,57 @@ func TestSystemE2E(t *testing.T) {
 		L2EngineAddrs: []string{endpoint(cfg.l2.nodeConfig)},
 	}
 	err = node.Run(context.Background())
+	require.Nil(t, err)
 	defer node.Close()
-	if err != nil {
-		t.Fatal(err)
-	}
+
+	// Deploy StateRootOracle
+	l2OutputPrivKey, err := bss.DerivePrivateKey(cfg.mnemonic, l2OutputHDPath)
+	require.Nil(t, err)
+	l2OutputAddr := crypto.PubkeyToAddress(l2OutputPrivKey.PublicKey)
+
+	nonce, err := l1Client.NonceAt(context.TODO(), l2OutputAddr, nil)
+	require.Nil(t, err)
+
+	opts, err := bind.NewKeyedTransactorWithChainID(
+		l2OutputPrivKey, big.NewInt(900),
+	)
+	require.Nil(t, err)
+	opts.Nonce = big.NewInt(int64(nonce))
+
+	submissionFrequency := big.NewInt(10) // 10 seconds
+	l2BlockTime := big.NewInt(2)          // 2 seconds
+	sroAddr, tx, stateRootOracle, err := sro.DeployStateRootOracle(
+		opts, l1Client, submissionFrequency, l2BlockTime, [32]byte{}, big.NewInt(0),
+	)
+	require.Nil(t, err)
+
+	_, err = txmgr.WaitMined(context.TODO(), l1Client, tx, time.Second, 1)
+	require.Nil(t, err)
+
+	initialSroTimestamp, err := stateRootOracle.LatestBlockTimestamp(&bind.CallOpts{})
+	require.Nil(t, err)
+
+	// Batch submitter
+	batchSubmitter, err := bss.NewBatchSubmitter(bss.Config{
+		BuildEnv:               "develop",
+		EthNetworkName:         "clique",
+		L1EthRpc:               httpHost(cfg.l1.nodeConfig.HTTPHost, cfg.l1.nodeConfig.HTTPPort),
```

Comment:
`endpoint(cfg.l1.nodeConfig)` returns the WS endpoint. It could be renamed to `wsEndpoint` and a `httpEndpoint(...)` could be added if the batch submitter prefers using HTTP

Reply:
i hadn't thought to try the bss with WS's, but it seems to work just fine. removed the `httpHost` logic